From fc7a6a5cbae6da60853738808c41a4caf0d46150 Mon Sep 17 00:00:00 2001
From: Terry Zhou <bjzhou@marvell.com>
Date: Wed, 20 Jul 2016 16:25:34 +0800
Subject: [PATCH 0627/2241] dts: usb: a3700: add regulator node for USB3 host
 VBUS

Add regulator node for USB3 VBUS, using IO-Expander0
@ 0x22 offset 1.

Change-Id: I06e241fca66abec2885d596a374f37ca61a9ba8f
Signed-off-by: Terry Zhou <bjzhou@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31272
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 8afd53d..ab1aa20 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -131,6 +131,7 @@
 			/* CON31 */
 			usb3@58000 {
 				status = "okay";
+				usb-phy = <&usb3_phy>;
 			};
 
 			usb@5e000 {
@@ -223,5 +224,20 @@
 			};
 		};
 	};
+
+	usb3_phy: usb3-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3_vbus>;
+	};
+
+	exp_usb3_vbus: usb3-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+	};
 };
 
-- 
2.7.4

